app/routes/upload.step1.tsx

```diff
@@ -0,0 +1,95 @@
+import { ActionFunctionArgs } from '@remix-run/node'
+import { Form, redirect, useNavigation } from '@remix-run/react'
+import { useState } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+import { Button } from '~/components/ui/button'
+import { supabase } from '~/supabaseClient'
+
+// Action to handle form submission
+export const action = async ({ request }: ActionFunctionArgs) => {
+  const body = await request.formData()
+
+  const file = body.get('cv-file') as File | null
+
+  if (file) {
+    const uniqueFileName = `${uuidv4()}-${file.name}`
+    const { data, error } = await supabase.storage
+      .from('resumes')
+      .upload(`public/${uniqueFileName}`, file) // this is just for testing now. will be secured later
+
+    if (error) {
+      console.error(error)
+    } else {
+      console.log('File uploaded successfully', data)
+    }
+  }
+  return redirect('/upload/step2')
+}
+
+const step1 = () => {
+  const [file, setFile] = useState<File | null>(null)
+
+  const navigation = useNavigation()
+  const isSubmitting = navigation.state === 'submitting'
+
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const uploadedFile = e.target.files?.[0] || null
+
+    if (uploadedFile && uploadedFile.size > 2 * 1024 * 1024) {
+      // 2MB limit
+      alert('File size exceeds the 2MB limit.') // TODO handle error display
+      return
```

Comment:
**suggestion**: Instead of using alert for error messages, consider using a more user-friendly approach, such as displaying the error message on the page.